resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/session/TestBug297635.java

```diff
@@ -45,26 +53,12 @@ public BundleContext getContext() {
 		return Platform.getBundle(PI_RESOURCES_TESTS).getBundleContext();
 	}
 
-	public void test1() {
+	public void testBug() throws Exception {
 		// install a bundle
-		try {
-			Bundle b = BundleTestingHelper.installBundle("1", getContext(), ContentTypeTest.TEST_FILES_ROOT + "content/bundle01");
-			BundleTestingHelper.resolveBundles(getContext(), new Bundle[] {b});
-			b.start(Bundle.START_TRANSIENT);
-		} catch (MalformedURLException e) {
-			fail("1.0", e);
-		} catch (BundleException e) {
-			fail("1.1", e);
-		} catch (IOException e) {
-			fail("1.2", e);
-		}
+		installBundle();
 
 		// register a save participant for the bundle
-		try {
-			getWorkspace().addSaveParticipant(BUNDLE01_ID, TestBug297635.this);
-		} catch (CoreException e) {
-			fail("2.0", e);
-		}
+		addSaveParticipant();
 
 		// create a project with a file
 		IProject project = getWorkspace().getRoot().getProject("Project1");
```

Comment:
You could think of separating the duplicated logic for creating a file and checking its existance into a method.

Reply:
Done! 
I extracted 2 private methods in 5fd81b6eee542d095e97e18f27ceffe337d070b6:
* `getProject`
* `createFileInProject`

The last one creates a file and makes sure it exists in the WS.